packages/fiber/src/native/polyfills.ts

```diff
@@ -38,22 +44,39 @@ export function polyfills() {
 
     // @ts-ignore
     texture.isDataTexture = true
-
-    getAsset(url)
-      .downloadAsync()
-      .then((asset: Asset) => {
-        texture.image = {
-          data: asset,
-          width: asset.width,
-          height: asset.height,
+    const asset = getAsset(url)
+
+    ;(async () => {
+      // this means it's android liniking
+      if (copyAsync && cacheDirectory && !asset.uri.includes('://')) {
+        const localUri = `${cacheDirectory}ExponentAsset-${asset.hash}.${asset.type}`
+        await copyAsync({
+          from: asset.uri,
+          to: localUri,
+        })
+        return {
+          data: { localUri },
+          asset,
```

Comment:
Does this only affect images/textures or does it also affect other assets like models? Might be best explained by https://github.com/facebook/react-native/issues/18216#issuecomment-412569517 which describes special casing for assets.